### PR TITLE
SW-3714 Make species view consistent with other interactions

### PR DESCRIPTION
**Pull request by alexTerraformation**
- Remove bulk action checkbox.
- Add species detail page
- Scientific Name column is a link that goes to detailed view of species.
- There’s an Edit Species button on species detail page.
- There is a Delete action (under “…” kebab) on species detail page.
- Add Species is a page rather than a modal

### CODE REVIEW

**Comment by chudo76** [comment]
@alexTerraformation A couple of minor changes if possible:

- Is it possible to have a grayed out checkbox for "Rare" instead of true/false? I think we do this for the Notifications checkbox under My Account:
image
- Can we show a comma delimited list for multiple Ecosystem Types for the Detailed view since we have room?
image

Otherwise looks good. Thank you!

File: src/scenes/Species/DeleteSpeciesModal.tsx [hunk 1]
```diff
@@ -46,60 +44,30 @@ export default function DeleteSpeciesDialog(props: DeleteSpeciesDialogProps): JS
...
     if (inUseSpecies) {
-      const forDeletion = speciesToDelete
-        .filter((species) => !inUseSpecies[species.id.toString()])
-        .map((species) => species.id);
-      const avoidDeletion = speciesToDelete
-        .filter((species) => inUseSpecies[species.id.toString()])
-        .map((species) => species.id);
-      setToDelete(forDeletion);
-      setCannotDelete(avoidDeletion);
+      if (inUseSpecies[speciesToDelete.id.toString()]) {
```
Comment (karthikbtf) [inline src/scenes/Species/DeleteSpeciesModal.tsx:47]: Nice refactor. This is fine, another option is disabling the delete button under the options menu with a tooltip on reason.

File: src/scenes/Species/SpeciesEditView.tsx [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
+import { Species } from 'src/types/Species';
+import { useOrganization } from 'src/providers/hooks';
+import { SpeciesService } from 'src/services';
+import TfMain from '../../components/common/TfMain';
+import { APP_PATHS } from 'src/constants';
+import { useHistory, useParams } from 'react-router-dom';
+import useDeviceInfo from 'src/utils/useDeviceInfo';
+import PageSnackbar from 'src/components/PageSnackbar';
+import useSnackbar from 'src/utils/useSnackbar';
+import PageForm from '../../components/common/PageForm';
```
Comment (karthikbtf) [inline src/scenes/Species/SpeciesEditView.tsx:13]: src/components

File: src/scenes/Species/SpeciesEditView.tsx [hunk 3]
```diff
@@ -0,0 +1,131 @@
...
+        </Box>
+        <Box
+          sx={{
+            backgroundColor: theme.palette.TwClrBg,
+            borderRadius: '32px',
+            padding: theme.spacing(3),
+            margin: 0,
+          }}
+        >
+          <SpeciesDetailsForm
```
Comment (karthikbtf) [inline src/scenes/Species/SpeciesEditView.tsx:119]: nice reuse

File: src/scenes/Species/index.tsx [hunk 4]
```diff
@@ -26,7 +31,22 @@ const SpeciesView = () => {
     return <EmptyStatePage pageName={'Species'} reloadData={reloadSpecies} />;
   }
 
-  return <SpeciesListView reloadData={reloadSpecies} species={species || []} />;
+  return (
```
Comment (karthikbtf) [inline src/scenes/Species/index.tsx:34]: 🎉

**Review by karthikbtf — COMMENTED** [review]
Great improvement 🙌

**Comment by alexTerraformation** [comment]
> * Can we show a comma delimited list for multiple Ecosystem Types for the Detailed view since we have room?

@chudo76 We have a lot of options for this so I think we should keep the mechanism for shortening the list.  I can extend the cutoff point to the width of the column maybe?

Screenshot 2024-02-14 at 11 27 02 AM

File: src/scenes/Species/DeleteSpeciesModal.tsx (hunk 1 above)
Comment (alexTerraformation) [reply src/scenes/Species/DeleteSpeciesModal.tsx:47]: I would be fine with that approach.  Keeps the logic out of the modal.  Do we have a design pattern for a disabled options menu option? @chudo76

Comment (karthikbtf) [reply src/scenes/Species/DeleteSpeciesModal.tsx:47]: One option is disabling the delete button but showing it under the options menu.
This looks like scope creep, we can do a follow up maybe. I noticed we don't have tooltip options on a button today (unless they are table/top-bar buttons).

Comment (alexTerraformation) [reply src/scenes/Species/DeleteSpeciesModal.tsx:47]: Good point.  Will enter a Jira to address this later.

**Review by karthikbtf — APPROVED** [review]
LGTM 🎉 
I understand there's a Chudo request outstanding.

File: src/scenes/Species/DeleteSpeciesModal.tsx (hunk 1 above)
Comment (chudo76) [reply src/scenes/Species/DeleteSpeciesModal.tsx:47]: > I would be fine with that approach. Keeps the logic out of the modal. Do we have a design pattern for a disabled options menu option? @chudo76

For Seed Bank Sub-locations, we just disable the button and have no info for why. For Projects we have a dialog box warning message (we don't prevent). At least with the dialog box we can let people know why they can't delete.

Dialog Box Title: "Delete Species"
Message: "This species is in use and cannot be deleted."
Button: "Got It"

File: src/scenes/Species/SpeciesDetailView.tsx [hunk 5]
```diff
@@ -88,6 +92,7 @@ export default function SpeciesDetailView(): JSX.Element {
       snackbar.toastError(strings.GENERIC_ERROR);
     }
     setDeleteSpeciesModalOpen(false);
+    reloadData();
```
Comment (karthikbtf) [inline src/scenes/Species/SpeciesDetailView.tsx:95]: reload data only on success?

File: src/scenes/Species/SpeciesEditView.tsx [hunk 6]
```diff
@@ -0,0 +1,131 @@
...
+  const gridSize = () => {
+    if (isMobile) {
+      return 12;
+    }
+    return 4;
+  };
+
+  const goToSpecies = (id?: number) => {
+    const speciesLocation = {
+      pathname: APP_PATHS.SPECIES + (id ? `/${id}` : ''),
```
Comment (karthikbtf) [inline src/scenes/Species/SpeciesEditView.tsx:48]: use APP_PATHS.SPECIES_DETAILS similar to L#48 SpeciesAddView.tsx

**Review by karthikbtf — APPROVED** [review]
Looks great to me! Thanks so much. A couple comments - ready to merge after that.

**Pull request merged**